### PR TITLE
Removal of spray dependency

### PR DESCRIPTION
**Pull request by barredijkstra**
Saw I still had #1 open.
This pull request removes the dependency on spray-util, the last non-provided dependency of the project.
Besides removing a transitive dependency with a fixed version, this would also allow for cross-compilation to Scala 2.12 (see #2).

If this pull request is accepted, I'll update my branch for #2 and send a subsequent pull request for that one.

### CODE REVIEW

**Comment by barredijkstra** [comment]
Amended pull request with changes to the travis config, reflecting the version changes in 1.0. This should also fix the build on master.

**Comment by codingismy11to7** [comment]
so... somehow I forgot about this PR when I went on a week-long spree updating all of our dependencies to build with 2.12. sorry to do this to you, but would you mind rebasing off of the wip_scala212 branch? I ask because I've been running off of that branch during 2.12 testing for a little while and am fairly confident that I didn't introduce new bugs. once you rebase if it looks good to you I'll go ahead and publish a non-snapshot release. feel free to replace anything I changed that doesn't look good - it was kind of a rush job.

while updating all the dependencies I was also making large arbitrary version bumps even though I wasn't changing the API (just a little implementation sometimes, and sometimes dropping Scala 2.10). So feel free to change the version back to a minor version bump if you think it's appropriate.

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,6 @@
 language: scala
 scala:
-  - 2.10.4
-  - 2.11.5
+  - 2.11.8
 jdk:
```
Comment (codingismy11to7) [inline .travis.yml]: I totally forgot about the Travis build. I guess it needs to run with JDK 8, IIRC... I think in my branch I told it to emit JDK6 class files for 2.11. guess 2.12.4 should be added too I suppose

**Comment by barredijkstra** [comment]
No problem, I will rebase the pull request and adjust accordingly.
Will probably take my 2.12 branch and rebase it against the wip_scala212 to see if there are any differences that might help out the wip_scala212.

Concerning the versions, given that the spray-util is the last non provided dependency I don't think there's much to change around.

**Comment by barredijkstra** [comment]
Small remark my side; is cross compiling Scala 2.11 against JDK6 instead of JDK7 intended?
I added it to the travis config for 2.11, but just making sure that it needs to be JDK6 of JDK7.

Also updated Scala 2.12 to 2.12.4 and 2.11 to 2.11.12 due to [CVE-2017-15288](http://scala-lang.org/news/security-update-nov17.html) 

Concerning the changes; routes taken by both of us for removing spray-util, etc. were functionally equal, so I used the ones from wip_scala212.

**Comment by codingismy11to7** [comment]
> is cross compiling Scala 2.11 against JDK6 instead of JDK7 intended?

yes, because 2.11 class files are JDK6, so anybody using 2.11 could be using that version (which describes us, we never took the time to adjust to JDK7 swing/jdbc changes, and jumped from 6 to 8)

**Comment by barredijkstra** [comment]
Other stuff being also on my plate, I now realise I rebased the pull off wip_scala212, but the pull request is still to master.
Do you want this pull request to stay on master (and effectively merging wip_scala212 as well) or you want it to be on wip_scala212?

**Comment by codingismy11to7** [comment]
master is good, thanks

**Review by codingismy11to7 — APPROVED** [review]

**Pull request merged**